Documentation/Global/Form/Index.rst

```diff
@@ -9,38 +9,92 @@ Form ViewHelper `<f:form>`
+
+The `Extbase controller <https://docs.typo3.org/permalink/t3coreapi:extbase-controller>`_
+action could for example look like this:
+
+..  literalinclude:: _codesnippets/_SearchController.php
+    :caption: packages/my_extension/Classes/Controller/SearchController.php
+
+..  _typo3-fluid-form-example-property:
+
+Property mapping - using the form with a model
+==============================================
+
+In most cases you will use the f:form ViewHelper with
+`Extbase models <https://docs.typo3.org/permalink/t3coreapi:extbase-model>`_ or
+data objects.
+
+For example, user could add a comment in such a form:
 
-A complex form with a specified encoding type
----------------------------------------------
+..  literalinclude:: _codesnippets/_CommentForm.html
+    :caption: packages/my_extension/Resources/Private/Templates/Comment/_CommentForm.html
 
-Form with enctype set::
+The Extbase Controller action displaying the form then creates the Domain object
+and passes it to the view. In the Fluid template above we use argument
+:ref:`object <t3viewhelper:viewhelper-argument-typo3-cms-fluid-viewhelpers-formviewhelper-object>`
+to pass any data the object might already contain to the ViewHelper.
 
-   <f:form action=".." controller="..." package="..." enctype="multipart/form-data">...</f:form>
+By using the argument "property" on the form input elements the properties of
+the model are automatically bound to the input elements.
 
-A Form which should render a domain object
-------------------------------------------
+The controller could look like this:
 
-Binding a domain object to a form::
+..  literalinclude:: _codesnippets/_CommentController.php
+    :caption: packages/my_extension/Classes/Controller/CommentController.php
 
-   <f:form action="..." name="customer" object="{customer}">
-      <f:form.hidden property="id" />
-      <f:form.textarea property="name" />
-   </f:form>
+If the model is not valid (see `Validation <https://docs.typo3.org/permalink/t3coreapi:extbase-validation>`_)
+Extbase will automatically refer the request back to the referring action
+(here commentFormAction()). By passing the object with the non-validated object
+to the view again, the user can see their faulty inputs and correct them instead
+of seeing an empty form.
 
-This automatically inserts the value of ``{customer.name}`` inside the
-textarea and adjusts the name of the textarea accordingly.
+..  _typo3-fluid-form-security:
+
+Security in Fluid forms
+=======================
+
+Fluid automatically adds a hidden field to forms, including an `__hmac`
+value. This value lists all allowed fields. If fields are added or
+removed via attacks, Extbase detects the mismatch and blocks submission.
+
+Form fields can be grouped in an array for efficient processing. The
+receiving action maps the data to a model, where validation occurs if
+rules are defined. Only valid data is passed to the action and stored in
+the database.
```

Comment:
Normally you can define newAction to show the form define as cached. BUT: If you work with a logged in user or show parts of the logged in user in the form you have to mark the newAction as uncached in ext_localconf.php. Else, it may happen that you see values from other users. Got a ticket from security team last year because of that.

Reply:
Would you like to make a follow up with some senteces about this? I really know nothing about it.